src/core/config/link_raw.h

```diff
@@ -81,6 +81,16 @@
 #define OPENTHREAD_CONFIG_SOFTWARE_CSMA_BACKOFF_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_SOFTWARE_TX_SECURITY_ENABLE
+ *
+ * Define to 1 if you want to enable software transmission security logic.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SOFTWARE_TX_SECURITY_ENABLE
+#define OPENTHREAD_CONFIG_SOFTWARE_TX_SECURITY_ENABLE 0
```

Comment:
Should this belong in `src/core/config/mac.h`? Also, should this name include `MAC` or `SUB_MAC`?

Reply:
This follows the similar behavior like `OPENTHREAD_CONFIG_SOFTWARE_CSMA_BACKOFF_ENABLE` and `OPENTHREAD_CONFIG_SOFTWARE_RETRANSMIT_ENABLE`. This flag is to indicate whether Tx security is done at OT(no matter which layer) or at platform(radio driver or hardware). If platform provide capability `OT_RADIO_CAPS_TRANSMIT_SEC`, then OT does not do so called software CSMA/retransmission/Tx security and will leave it to platform(considered as hardware CSMA/Retx/Tx Sec). If not, OT will do this.

This flag is configured per platform, so it resides in platform's openthread-core-xxx-config.h.

Reply:
I think the current pattern is out of date. These options were created in #1248 and specific to LINK_RAW feature at the time. Now these features have been combined into the sub-MAC in #3262 and included as part of RCP, independent of whether LINK_RAW features is enabled. As a result, I think these options are likely better scoped under the MAC component.

Reply:
I see. I will move all these flags to `src/core/config/mac.h`.